common/optimiser/Optimiser.java

```diff
@@ -45,30 +53,49 @@ public class Optimiser {
     private MPSolver solver;
     private MPSolverParameters parameters;
     private Status status;
-    private boolean hasSolver;
-    private double objectiveValue;
+    private boolean isConstraintsUpToDate;
+    private Double objectiveValue;
```

Comment:
`objectiveValue` may not exist yet until we have actually got some objective function. A nullable object makes more sense untl a objective is defined